File: VPSApps/old/php/qotd.php
<?php
$_inBold = false;
$_inItalic = false;
$_qotd = "";
$_result = array();

function qotd_get($qotd_file)
{
	global $_result;

	$content = file($qotd_file);
	_parse($content);
	reset($_result);
	return $_result[rand(0, count($_result))];
}

function _parse($content)
{
	global $_qotd;

	reset($content);
	$length = count($content);
	foreach ($content as $line) {
		$line = trim($line);
		if (_is_marker($line)) {
			_add_qotd();
			continue;
		}

		$s = _parse_line($line);
		$_qotd .= $s;
		$_qotd .= '<br />';
	}

	_add_qotd();
}

function _is_marker($line)
{
	return "%" === $line;
}

function _add_qotd()
{
	global $_qotd, $_result;

	$html = trim($_qotd);
	if (strlen($html) < 1) {
		return;
	}

	$_result[] .= $_qotd;
	$_qotd = "";
}

function _parse_line($line)
{
	global $_inBold, $_inItalic;

	$length = strlen($line);
	$text = "";

	for ($index = 0; $index < $length; $index++) {
		switch ($line[$index]) {
			case '\\':
				$index++;
				$text .= $line[$index];
				break;

			case '*':
				$text .= $_inBold ? "</b>" : "<b>";
				$_inBold = !$_inBold;
				break;

			case '_':
				$text .= $_inItalic ? "</i>" : "<i>";
				$_inItalic = !$_inItalic;
				break;

			case '#':
				$text .= '&nbsp;';
				break;

			default:
				$text .= $line[$index];
				break;
		}
	}

	return $text;
}

$qotd = htmlentities(qotd_get('qotds.txt'));
$ts = date('YmdHHiiss');
$l_time = localtime(time(), true);
$time = mktime(0, 0, 0, intval($l_time["tm_mon"]) + 1, $l_time["tm_mday"], intval($l_time["tm_year"]) + 1900);
$pub_date = date('r', $time);
$rss = <<<EOT
<?xml version="1.0" encoding="utf-8"?>
<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom">
<channel>
	<title>Quote of the Day</title>
	<link>https://www.schau.dk/qotd.php</link>
	<description>Quote of the Day!</description>
	<atom:link href="https://www.schau.dk/qotd.php" rel="self" type="application/rss+xml" />
	<ttl>240</ttl>
	<pubDate>$pub_date</pubDate>
	<item>
		<title>Quote of the Day</title>
		<link>https://www.schau.dk/qotd.php?ts=$ts</link>
		<description>$qotd</description>
		<guid>https://www.schau.dk/qotd.php?ts=$ts</guid>
	</item>
</channel>
</rss>
EOT;

header("Content-Type: application/rss+xml");
echo $rss;
?>

File: VPSApps/old/python/python/qotd.py
""" qotd """
import random
import os
import platform
from vpsutils import VpsUtils


def get_qotd(vps):
    """ Read random quote from quote file.
    	Arguments:
        	vps. VpsUtils
    """
    name = "_qotd" if platform.system() == 'Windows' else ".qotd"
    qotd_file = os.path.join(vps.home, name)

    with open(qotd_file, 'r') as qotds_file:
        content = qotds_file.read()

    qotds = content.split('%\n')

    qotd = ''
    while qotd.strip() == '':
        qotd = random.choice(qotds)

    return qotd


def main():
    """ Main entry method. """
    vps = VpsUtils('Quote of the Day')
    qotd = get_qotd(vps)

    index = 0
    str_len = len(qotd)
    in_bold = False
    in_italic = False
    while index < str_len:
        character = qotd[index]
        if character == '\\':
            index = index + 1
            vps.html_add(qotd[index])
        elif character == '\n':
            vps.html_add('<br />')
        elif character == '*':
            vps.html_add('</b>' if in_bold else '<b>')
            in_bold = not in_bold
        elif character == '_':
            vps.html_add('</i>' if in_italic else '<i>')
            in_italic = not in_italic
        elif character == '#':
            vps.html_add('&nbsp;')
        else:
            vps.html_add(character)

        index = index + 1

    vps.deliver()


if __name__ == '__main__':
    main()
